### ISSUE TITLE
Question about _make_train_function->train_function->function

### ISSUE THREAD

**Issue by stavBodik** [keras-team/keras#8217]
Hello ,
I am new to this framework and python at all...I am looking for the source code where the loss is actually calculated.
As I understand,  _make_train_function creates some function which is later runs on GPU device in parallel .
Where can I find the implementation of 'train_function' please? 
I think I am looking for the backend function which is written not in python ? : keras.backend.tensorflow_backend.Function object at 0x00000000103B3B38

Or how its all works ?
Where can I find documentation or explanation for this source codes ? I mean more then comments which explains steps/functions.

 Thanks !

`
self.train_function = K.function(inputs,
                                             [self.total_loss] + self.metrics_tensors,
                                             updates=updates,
                                             name='train_function',
                                             **self._function_kwargs)
`
